yb-voyager/cmd/importDataFileTaskImporter_test.go

```diff
@@ -278,3 +286,393 @@ func TestTaskImportErrorsOutWithAbortErrorPolicy(t *testing.T) {
 		workerPool.Wait()
 	})
 }
+
+// ------------------------------------------------------------------------------------------------
+// Import Batch Retryable Error Detection Tests
+// ------------------------------------------------------------------------------------------------
+
+type testCase struct {
+	name              string
+	errorType         string
+	expectedRetryable bool
+	description       string
+	dataGenerator     func() string
+}
+
+func createTestCases() []testCase {
+	return []testCase{
+		{
+			name:              "numeric_value_out_of_range",
+			errorType:         "22003",
+			expectedRetryable: false,
+			description:       "Should not retry numeric value out of range errors",
+			dataGenerator:     createNumericValueOutOfRangeData,
+		},
+		{
+			name:              "string_data_right_truncation",
+			errorType:         "22001",
+			expectedRetryable: false,
+			description:       "Should not retry string data right truncation errors",
+			dataGenerator:     createStringDataRightTruncationData,
+		},
+		{
+			name:              "unique_violation",
+			errorType:         "23505",
+			expectedRetryable: false,
+			description:       "Should not retry unique violation errors",
+			dataGenerator:     createUniqueViolationData,
+		},
+		{
+			name:              "check_violation",
+			errorType:         "23514",
+			expectedRetryable: false,
+			description:       "Should not retry check violation errors",
+			dataGenerator:     createCheckViolationData,
+		},
+		{
+			name:              "invalid_text_representation",
+			errorType:         "22P02",
+			expectedRetryable: false,
+			description:       "Should not retry invalid text representation errors",
+			dataGenerator:     createInvalidTextRepresentationData,
+		},
+		{
+			name:              "bad_copy_file_format",
+			errorType:         "22P04",
+			expectedRetryable: false,
+			description:       "Should not retry bad copy file format errors",
+			dataGenerator:     createBadCopyFileFormatData,
+		},
+		{
+			name:              "valid_data",
```

Comment:
can we add a case wherein it throws an error (that is not part of 22/23), so we deem it to be retryable. 

This might be hard to do, but can we explore it a little first?